### PR TITLE
homework-6

### PR DESCRIPTION
**Pull request by EvgeniyaMarchenkova**

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -17,57 +25,83 @@ export interface Handlers {
...
-    selectMovie,
-  } = useMovieService();
+  const movies = useAppSelector(selectMovies);
+  const selectedMovie = useAppSelector(selectSelectedMovie);
 
   const [searchText, setSearchText] = useState('');
+  const [sortBy, setSortBy] = useState('');
   const [modalWindowType, setModalWindowType] = useState(ModalWindowType.None);
   const [isViewMode, setIsViewMode] = useState(false);
+  const [selectedGenre, setSelectedGenre] = useState();
```
Comment (AlexGusew) [inline src/components/App.tsx:30]: By the AC filtering by genre should be in the redux

File: src/components/App.tsx [hunk 2]
```diff
@@ -17,57 +25,83 @@ export interface Handlers {
...
+    search: searchText,
+    searchBy: SearchBy.Title,
+    sortOrder: SortOrder.Increase,
+    sortBy: sortBy,
+    filter: selectedGenre,
+  } as const;
+  const { data, error, isLoading, refetch } = useGetMoviesQuery(queryParams);
+
+  const dispatch = useAppDispatch();
+  dispatch(setMovies(data?.data));
```
Comment (AlexGusew) [inline src/components/App.tsx]: Dispatch can be situated only in event handlers or useEffects

File: src/components/App.tsx [hunk 3]
```diff
@@ -17,57 +25,83 @@ export interface Handlers {
...
+
+  const { editMovie, addMovie, deleteMovie } = useMovieService();
```
Comment (faustAbc) [inline src/components/App.tsx:49]: Regarding AC, This is intended to be in the redux

File: src/components/App.tsx (hunk 2 above)
Comment (EvgeniyaMarchenkova) [reply src/components/App.tsx]: Fixed

File: src/components/MainContent.tsx [hunk 4]
```diff
@@ -1,30 +1,28 @@
...
-        onChangeSelectedGenre={setSelectedGenre}
+        selectedGenre={props.selectedGenre}
+        onChangeSort={props.onChangeSort}
+        onChangeSelectedGenre={props.onChangeSelectedGenre}
       />
       <MoviesGrid
-        movies={props.movies}
-        genre={selectedGenre}
-        searchText={props.searchText}
+        movies={props.movies || []}
```
Comment (AlexGusew) [inline src/components/MainContent.tsx:24]: It's better to add default value to MoviesGrid rather then to add `|| []` every time

File: src/components/NavBar.tsx [hunk 5]
```diff
@@ -24,21 +24,32 @@ const MenuItem = styled.span<MenuItemProps>`
...
   Genre.Documentary,
   Genre.Comedy,
   Genre.Crime,
   Genre.Horror,
+  Genre.Adventure,
+  Genre.Animation,
+  Genre.Drama,
+  Genre.Family,
+  Genre.Romance,
 ] as const;
```
Comment (AlexGusew) [inline src/components/NavBar.tsx:41]: Alternatively, you can use Object.values(Genre);

**Review by AlexGusew — APPROVED** [review]

File: src/components/MainContent.tsx (hunk 4 above)
Comment (EvgeniyaMarchenkova) [reply src/components/MainContent.tsx:24]: Fixed.

**Pull request merged**